src/include/migraphx/dom_info.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_DOM_INFO_HPP
+#define MIGRAPHX_GUARD_RTGLIB_DOM_INFO_HPP
+#include <migraphx/common_header.hpp>
+#include <migraphx/set_operator.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+struct dom_info
```

Comment:
Can we have some documentation on the methods of this class? Also, it would be good to have some test cases for this class as well.

Reply:
This is invoked from memory coloring.  I added a test to memory_coloring_test.cpp which already triggers codes in dom_info.